packages/grafana-data/src/transformations/transformers/filterByValue.ts

```diff
@@ -0,0 +1,131 @@
+  valueFilters: ValueFilter[];
+  type: string; // 'include' or 'exclude'
+  match: string; // 'all' or 'any'
+}
+
+export const filterByValueTransformer: DataTransformerInfo<FilterByValueTransformerOptions> = {
+  id: DataTransformerID.filterByValue,
+  name: 'Filter by value',
+  description: 'Filter the data points (rows) depending on the value of certain fields',
+  defaultOptions: {
+    valueFilters: [{ fieldName: null, filterExpression: null, filterType: ValueFilterID.regex }],
+    type: 'include',
+    match: 'all',
+  },
+
+  /**
+   * Return a modified copy of the series.  If the transform is not or should not
+   * be applied, just return the input series
+   */
+  transformer: (options: FilterByValueTransformerOptions) => {
+    const includeRow = options.type === 'include';
+    const matchAll = options.match === 'all';
+
+    return (data: DataFrame[]) => {
+      if (options.valueFilters.length === 0) {
+        return data;
+      }
+
+      const processed: DataFrame[] = [];
+
+      let includeThisRow = []; // All data points will be flagged for include (true) or exclude (false) in this variable
+
+      for (let frame of data) {
+        for (let filterIndex = 0; filterIndex < options.valueFilters.length; filterIndex++) {
+          let filter = options.valueFilters[filterIndex];
+
+          // Find the matching field for this filter
+          let field = null;
+          for (let f of frame.fields) {
+            if (getFieldDisplayName(f) === filter.fieldName) {
+              field = f;
+              break;
+            }
+          }
+
+          if (field === null) {
+            continue; // No field found for for this filter in this frame, ignore
+          }
+
+          // This creates the filter instance we need (with the test function) we need to match the rows
+          let filterInstance = valueFiltersRegistry.get(filter.filterType).getInstance({
+            filterExpression: filter.filterExpression,
+            fieldType: field.type,
+          });
+
+          if (!filterInstance.isValid) {
+            continue;
+          }
+
+          if (matchAll) {
+            // Run the test on each row
+            for (let row = 0; row < frame.length; row++) {
+              if (!filterInstance.test(field.values.get(row))) {
+                includeThisRow[row] = !includeRow;
+              } else if (filterIndex === 0) {
+                includeThisRow[row] = includeRow;
+              }
+            }
+          } else {
+            // Run the test on each row
+            for (let row = 0; row < frame.length; row++) {
+              if (filterInstance.test(field.values.get(row))) {
+                includeThisRow[row] = includeRow;
+              } else if (filterIndex === 0) {
+                includeThisRow[row] = !includeRow;
+              }
+            }
+          }
+        }
+
+        // Create the skeleton of the new data, copy original field attributes
+        let filteredFields: Field[] = [];
```

Comment:
```suggestion
        let filteredFields: MutableField[] = [];
```

If you change from a `Field` to a `MutableField` you will resolve the typing error that currently exists in this branch. You also need to import the `MutableField` by adding `import { MutableField } from '../../dataframe';` to the top of the file.